docs/sources/k6/next/testing-guides/performance-testing-grpc-services.md

```diff
@@ -0,0 +1,206 @@
+
+#### Client streaming
+
+The client streaming mode is the opposite of the server streaming mode. The client sends multiple requests to the server, which in turn replies with a single response.
+
+![Client streaming](/media/docs/k6-oss/diagram-grpc-client.png)
+
+#### Bi-directional streaming
+
+In bi-directional streaming mode, both the client and the server may send multiple messages.
+
+![Bi-directional streaming](/media/docs/k6-oss/diagram-grpc-bidirectional.png)
+
+### The `.proto` definition
+
+The messages and services used for gRPC are described in .proto files, containing [Protocol buffers](https://en.wikipedia.org/wiki/Protocol_Buffers), or protobuf, definitions.
+
+The definition file is then used to generate code which can be used by both senders and receivers as a contract for communicating through these messages and services. As the binary format used by gRPC lacks any self-describing properties, this is the only way for senders and receivers to know how to interpret the messages.
+
+For this guide, you can use the `quickpizza.proto` definition available for download on the [QuickPizza demo application GitHub repository](https://raw.githubusercontent.com/grafana/quickpizza/refs/heads/main/proto/quickpizza.proto). For details on how to build your own gRPC proto definition, refer to [the official gRPC docs](https://grpc.io/docs/what-is-grpc/core-concepts/).
+
+```protobuf
+syntax = "proto3";
+option go_package = "pkg/grpc/quickpizza";
+package quickpizza;
+
+service GRPC {
+    rpc Status(StatusRequest) returns (StatusResponse) {}
+    rpc RatePizza(PizzaRatingRequest) returns (PizzaRatingResponse) {}
+}
+
+message StatusRequest {
+}
+
+message StatusResponse {
+    bool ready = 1;
+}
+
+message PizzaRatingRequest {
+    repeated string ingredients = 1;
+    string dough = 2;
+}
+
+message PizzaRatingResponse {
+    int32 stars_rating = 1;
+}
+```
+
+## Write a gRPC performance test with k6
+
+Starting with k6 v0.29.0, you can use a built-in module for gRPC communication. You can find more details for all the methods available on the [k6/net/grpc module documentation](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-net-grpc/).
+
+### Create the test
+
+The gRPC module is a separate module, available from your test script as `k6/net/grpc`. Before you can use it, you first have to create an instance of the client. Instantiating the client, as well as the `.load` operation, is only available during test initialization, that is, directly in the global scope.
+
+<!-- md-k6:skip -->
+
+```javascript
+import grpc from 'k6/net/grpc';
+import { check, sleep } from 'k6';
+
+const client = new grpc.Client();
+```
+
+Next, load a `.proto` definition applicable for the system under test. For the purpose of this article, you can use [QuickPizza](grpc-quickpizza.grafana.com:443). The `.load()` function takes two arguments, the first one being an array of paths to search for proto files, and the second being the name of the file to load.
```

Comment:
Clicking on the QuickPizza link won't work for every user. Should we add a notice?